src/prefect/cli/execute.py

```diff
@@ -66,6 +66,12 @@ def flow_run():
         with prefect.context(secrets=secrets, loading_flow=True):
             flow = storage.get_flow(storage.flows[flow_data.name])
 
+        # Load run config from backend if not present on flow object
+        if flow.run_config is None and flow_data.run_config:
+            flow.run_config = prefect.serialization.run_config.RunConfigSchema().load(
+                flow_data.run_config
+            )
+
         with prefect.context(secrets=secrets):
             if getattr(flow, "run_config", None) is not None:
```

Comment:
Since the only reason we need the `run_config` in execution is to determine if the flow is using a legacy `environment` based configuration, we should be able to drop the load (and mutation) above and just modify the boolean condition here. This will also play nicer with future changes where the run-config info can be overridden in the backend (if we let legacy flows be overridden). Something like:

```suggestion
            if flow_data.run_config is not None:
```